### PR TITLE
clippy updates for rust 1.47

### PR DESCRIPTION
**Pull request by bluejekyll**

### CODE REVIEW

**Comment by bluejekyll** [comment]
> I think this is wrong? The current expression is effectively max(min(length, 512), 4096), which always evaluates to 4096.

yeah, I need to fix that. I didn't have time to think through the clippy warning yesterday.. it should invert the max and min...

File: crates/client/src/rr/dnssec/keypair.rs [hunk 1]
```diff
@@ -407,10 +407,9 @@ impl<K: HasPrivate> KeyPair<K> {
                                 }
                                 part = &part[1..];
                             }
-                            for _ in 0..(part_len - part.len()) {
-                                // Pad with zeros. All numbers are big-endian here.
-                                ret.push(0x00);
-                            }
+
+                            // Pad with zeros. All numbers are big-endian here.
+                            ret.resize(part_len - part.len(), 0x00);
```
Comment (djc) [inline crates/client/src/rr/dnssec/keypair.rs:412]: What are the intended semantics here? Note that resize() takes new_len, not the amount of items to add.

File: crates/https/src/https_server.rs [hunk 2]
```diff
@@ -66,6 +66,7 @@ pub(crate) async fn message_from_post<R>(
 where
     R: Stream<Item = Result<Bytes, h2::Error>> + 'static + Send + Debug + Unpin,
 {
+    #[allow(clippy::min_max)]
```
Comment (djc) [inline crates/https/src/https_server.rs:69]: I suppose you missed my comments at https://github.com/bluejekyll/trust-dns/commit/d53381edda7e2c787789b4d5704724cc5f24199b? I think this is wrong.

File: crates/proto/src/rr/domain/name.rs [hunk 3]
```diff
@@ -1450,6 +1450,7 @@ mod tests {
     }
 
     #[test]
+    #[allow(clippy::same_item_push)]
```
Comment (djc) [inline crates/proto/src/rr/domain/name.rs:1453]: Also see aforementioned link for an alternative suggestion.

**Review by djc — COMMENTED** [review]
I think some things could be improved?

File: crates/client/src/rr/dnssec/keypair.rs (hunk 1 above)
Comment (bluejekyll) [reply crates/client/src/rr/dnssec/keypair.rs:412]: hm, I might have gotten that conversion wrong. Luckily the tests also appear to have caught it.

File: crates/https/src/https_server.rs (hunk 2 above)
Comment (bluejekyll) [reply crates/https/src/https_server.rs:69]: I got it. I have a patch locally...

File: crates/proto/src/rr/domain/name.rs (hunk 3 above)
Comment (bluejekyll) [reply crates/proto/src/rr/domain/name.rs:1453]: yup, I have this

`bytes.extend((0..=n).into_iter().map(|_| n));`

**Comment by bluejekyll** [comment]
> bytes.extend(iter::repeat(n).take(n as usize));

That's a little cleaner than what I did in that context... preference?

**Comment by djc** [comment]
> > bytes.extend(iter::repeat(n).take(n as usize));
> 
> That's a little cleaner than what I did in that context... preference?

Yeah, I think I prefer my take, too.

File: crates/client/src/rr/dnssec/keypair.rs [hunk 4]
```diff
@@ -407,10 +407,9 @@ impl<K: HasPrivate> KeyPair<K> {
                                 }
                                 part = &part[1..];
                             }
-                            for _ in 0..(part_len - part.len()) {
-                                // Pad with zeros. All numbers are big-endian here.
-                                ret.push(0x00);
-                            }
+
+                            // Pad with zeros. All numbers are big-endian here.
+                            ret.extend((0..(part_len - part.len())).into_iter().map(|_| 0x00));
```
Comment (djc) [inline crates/client/src/rr/dnssec/keypair.rs:412]: Should this also get the iter::repeat() treatment?

Comment (bluejekyll) [reply crates/client/src/rr/dnssec/keypair.rs:412]: yea, I can do that. I can do it in about an hour.

**Review by djc — APPROVED** [review]

**Pull request merged**